src/main/kotlin/the/coding/force/exploring_kotlin_coroutines/service/coroutine/DeleteDataServiceCoroutine.kt

```diff
@@ -0,0 +1,32 @@
+package the.coding.force.exploring_kotlin_coroutines.service.coroutine
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import the.coding.force.exploring_kotlin_coroutines.Exception.DataNotFoundException
+import the.coding.force.exploring_kotlin_coroutines.repository.DataRepository
+
+@Service
+class DeleteDataServiceCoroutine(
+    private val dataRepository: DataRepository,
+    private val ioDispatcher: CoroutineDispatcher
+) {
+    private val logger = KotlinLogging.logger { }
+
+    suspend fun delete(dataId: Long) {
+        withContext(ioDispatcher) {
+            dataRepository.findById(dataId).ifPresentOrElse(
+                {
+                    data ->
+                    dataRepository.deleteById(dataId)
+                    logger.info { "DeleteDataService.delete: Data with ID ${data.id} was deleted" }
```

Comment:
mesmo comentário de cima

Reply:
commit: [improving logs, change Dispatcher strategy, remove webFlux dependency](https://github.com/the-coding-force/exploring-kotlin-coroutines__backend__spring-kotlin__poc/pull/13/commits/bcb1229dfa29acc3455c10a16adc4b18b4edd81d)